lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -1,23 +1,37 @@
 require 'fastlane_core/ui/ui'
 require 'octokit'
 require 'open-uri'
+require 'securerandom'
 
 module Fastlane
   UI = FastlaneCore::UI unless Fastlane.const_defined?('UI')
 
   module Helper
     class GithubHelper
+      def self.github_token!
+        token = [
+          'GHHELPER_ACCESS', # For historical reasons / backward compatibility
+          'GITHUB_TOKEN',    # Used by the `gh` CLI tool
+        ].map { |key| ENV[key] }
+                .compact
+                .first
+
+        token || UI.user_error!('Please provide a GitHub authentication token via the `GITHUB_TOKEN` environment variable')
+      end
+
       def self.github_client
-        client = Octokit::Client.new(access_token: ENV['GHHELPER_ACCESS'])
+        @@client ||= begin
+          Octokit::Client.new(access_token: github_token!)
 
-        # Fetch the current user
-        user = client.user
-        UI.message("Logged in as: #{user.name}")
+          # Fetch the current user
+          user = client.user
+          UI.message("Logged in as: #{user.name}")
 
-        # Auto-paginate to ensure we're not missing data
-        client.auto_paginate = true
+          # Auto-paginate to ensure we're not missing data
+          client.auto_paginate = true
+        end
 
-        client
+        @@client
```

Comment:
In ruby, assignments like `a = b` or `a ||= b` already return the assigned value.

That's the whole beauty of the `@foo ||= begin … end` construct as you only have to wrap the whole method in that without having to add anything else at the end after that wrapping to make the memoization work.

So you should remove this last `@@client` line and let the whole `@@client ||= begin … end` block be the return value itself.

_(I'm not creating a GitHub suggestion for it because you'll also need to remove the empty lines above it to not make `rubocop` complain, but GitHub does not support `suggestion` blocks on deleted lines…)_

Reply:
TIL, thanks!

Addressed in https://github.com/wordpress-mobile/release-toolkit/pull/313/commits/f495914d48b63dd71437b65f35b85143c18f5941